model/trigger/task_jira.go

```diff
@@ -0,0 +1,292 @@
+package trigger
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/model/build"
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/evergreen-ci/evergreen/model/version"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+// DescriptionTemplateString defines the content of the alert ticket.
+const descriptionTemplateString = `
+h2. [{{.Task.DisplayName}} failed on {{.Build.DisplayName}}|{{.UIRoot}}/task/{{.Task.Id}}/{{.Task.Execution}}]
+Host: [{{.Host.Host}}|{{.UIRoot}}/host/{{.Host.Id}}]
+Project: [{{.Project.DisplayName}}|{{.UIRoot}}/waterfall/{{.Project.Identifier}}]
+Commit: [diff|https://github.com/{{.Project.Owner}}/{{.Project.Repo}}/commit/{{.Version.Revision}}]: {{.Version.Message}}
+{{range .Tests}}*{{.Name}}* - [Logs|{{.URL}}] | [History|{{.HistoryURL}}]
+{{end}}
+`
+const (
+	jiraFailingTasksField     = "customfield_12950"
+	jiraFailingTestsField     = "customfield_15756"
+	jiraFailingVariantField   = "customfield_14277"
+	jiraEvergreenProjectField = "customfield_14278"
+	jiraFailingRevisionField  = "customfield_14851"
+	jiraMaxTitleLength        = 254
+)
+
+// supportedJiraProjects are all of the projects, by name that we
+// expect to be compatible with the custom fields above.
+var supportedJiraProjects = []string{"BFG", "BF", "EVG", "MAKE", "BUILD"}
```

Comment:
can we push supported jira projects and the custom fields into some global configuration? 

this code from the legacy alerts was a hack predicated on the thought that we'd be able to get out of it eventually.